main.tf

```diff
@@ -1,136 +1,179 @@
+resource "aws_security_group" "rdp" {
+  description                         = "Allow incoming rdp traffic from admin_ips"
+  vpc_id                              = "${var.vpc_id}"
+
+      tags = {
+        Name                          = "allow_rdp"
+      }
+      ingress {
+        from_port                     = 3389
+        to_port                       = 3389
+        protocol                      = "tcp"
+      cidr_blocks                     = ["${var.admin_ips}"]
+            }
+      ingress {
+        from_port                     = 3389
+        to_port                       = 3389
+        protocol                      = "udp"
+      cidr_blocks                     = ["${var.admin_ips}"]
+            }
+      ingress {
+        from_port                     = 0
+        to_port                       = 65535
+        protocol                      = "tcp"
+      cidr_blocks                     = ["0.0.0.0/0"]
+            }
+    }
+
+resource "aws_instance" "mesos-agent-windows" {
+  count                               = "${var.num_winagent}"
+  ami                                 = "${data.aws_ami.winAmi.id}"
+  instance_type                       = "t3.xlarge"
+  get_password_data                   = "true"
+  root_block_device {
+    volume_type                       = "gp2"
+    volume_size                       = 230
+    }
+  tags = {
+    Name                              = "${var.cluster_name}-winagent-${count.index}"
+    expiration                        = "${var.expiration}"
+    owner                             = "${var.owner}"
   }
-
-  os = "${var.dcos_instance_os}"
+  key_name                            = "${var.aws_key_name}"
+  subnet_id                           = "${element(var.subnet_id,0)}"
+  security_groups                     = ["${list(
+      var.security_group_internal,
+      var.security_group_admin,
+      aws_security_group.rdp.id)}"]
+
+  user_data                           = <<-EOF
+  <script>
+    winrm quickconfig -q & winrm set winrm/config @{MaxTimeoutms="1800000"} & winrm set winrm/config/service @{AllowUnencrypted="true"} & winrm set winrm/config/service/auth @{Basic="true"}
+  </script>
+  <powershell>
+  New-SelfSignedCertificate -DnsName $(Invoke-RestMethod -uri http://169.254.169.254/latest/meta-data/local-hostname) -CertStoreLocation Cert:\LocalMachine\My 
+  New-Item WSMan:\localhost\Listener -Address * -Transport HTTPS -HostName $(Invoke-RestMethod -uri http://169.254.169.254/latest/meta-data/local-hostname) -CertificateThumbPrint $(ls Cert:\LocalMachine\My).Thumbprint -Force
+  Set-NetFirewallProfile -Profile Domain,Public,Private -Enabled False
+  Set-MpPreference -DisableRealtimeMonitoring $true
+
+  </powershell>
+                    EOF
 }
 
-resource "aws_instance" "instance" {
-  instance_type = "${var.instance_type}"
-  ami           = "${coalesce(var.ami, module.dcos-tested-oses.aws_ami)}"
+resource "null_resource" "run_ansible_from_bootstrap_node_to_install_dcos" {
```

Comment:
this should not be part of the instance module. We strictly separate between infrastructure and installation layer